pkg/oc/cli/cmd/set/volume.go

```diff
@@ -387,44 +359,81 @@ func (v *VolumeOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, out,
 	v.Cmd = cmd
 	v.DefaultNamespace = cmdNamespace
 	v.ExplicitNamespace = explicit
-	v.Out = out
-	v.Err = errOut
 	v.Mapper = mapper
 	v.Typer = typer
 	v.CategoryExpander = f.CategoryExpander()
 	v.RESTClientFactory = f.ClientForMapping
 	v.UpdatePodSpecForObject = f.UpdatePodSpecForObject
 	v.Encoder = f.JSONEncoder()
 
+	// Complete AddOpts
+	if v.Add {
+		err := v.AddOpts.Complete()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *AddVolumeOptions) Complete() error {
 	// In case of volume source ignore the default volume type
-	if len(v.AddOpts.Source) > 0 {
-		v.AddOpts.Type = ""
+	if len(a.Source) > 0 {
```

Comment:
since Validate used to be called first, and this block was called in Complete, move this block to the end of this method